src/analysis/normed_space/operator_norm.lean

```diff
@@ -16,9 +16,89 @@ particular
 import algebra.module
 import analysis.normed_space.bounded_linear_maps
 
-variables {k : Type*} 
-variables {E : Type*} {F : Type*}
-
+variable  {k : Type*}
+variables {E : Type*} {F : Type*} {G : Type*}
+
+-- refactor : formulation for bundled bounded_linear_map
+section op_norm
+
+variable  [normed_field k]
+variables [normed_space k E] [normed_space k F] [normed_space k G]
+
+open bounded_linear_map
+
+variables (α : k) {x : E}
+variables (f g : bounded_linear_map k E F) (h : bounded_linear_map k F G)
+
+noncomputable def op_norm  := real.Inf { c | c ≥ 0 ∧ ∀ x, ∥f x∥ ≤ c * ∥x∥ }
+
+noncomputable instance : has_norm (bounded_linear_map k E F) := ⟨op_norm⟩
+
+-- so that invocations of real.Inf_le make sense:
+lemma bounds_nonempty {f : bounded_linear_map k E F}:
+  ∃ c, c ∈ { c | c ≥ 0 ∧ ∀ x, ∥f x∥ ≤ c * ∥x∥ } :=
+  let ⟨M, hMp, hMb⟩ := has_pos_bound in ⟨M, le_of_lt hMp, hMb⟩
+lemma bounds_bdd_below {f : bounded_linear_map k E F}:
+  bdd_below { c | c ≥ 0 ∧ ∀ x, ∥f x∥ ≤ c * ∥x∥ } :=
+  ⟨0, λ _ ⟨hn, _⟩, hn⟩
+
+lemma op_norm_nonneg: ∥f∥ ≥ 0 :=
+  real.lb_le_Inf _ bounds_nonempty (λ _ ⟨hx, _⟩, hx)
+
+lemma le_op_norm: ∥f x∥ ≤ ∥f∥ * ∥x∥ :=
+  or.elim (eq_or_lt_of_le (norm_nonneg x))
+  (λ heq, by rw [←heq, mul_zero,
+    (norm_eq_zero _).1 heq.symm, map_zero, norm_zero])
+  (λ hlt, le_mul_of_div_le hlt
+    ((real.le_Inf _ bounds_nonempty bounds_bdd_below).2
+    (λ c ⟨_, hc⟩, div_le_of_le_mul hlt (by rw mul_comm; exact hc _))))
+
+-- results about bounding the unit ball. (naming conventions?)
+lemma ratio_le_op_norm: ∥x∥ > 0 → ∥f x∥ / ∥x∥ ≤ ∥f∥ :=
```

Comment:
Again, the assumption is useless, right?

Reply:
fixed in 8946ddcaee6ff2c15022befd6b1668cb11c2fca5 .